activemodel/lib/active_model/attribute_set/builder.rb

```diff
@@ -76,15 +76,17 @@ def ==(other)
     end
 
     def marshal_dump
-      materialize
+      [@types, @values, @additional_types, @default_attributes, @delegate_hash]
```

Comment:
> This will lose any mutations that have been made to the record. Can we just stick `@delegate_hash` in here and lose the materialized branch entirely?

Good point... I've fixed in 7c7e7b7.